src/sentry/search/django/backend.py

```diff
@@ -136,12 +136,22 @@ def _build_queryset(self, project, query=None, status=None, tags=None,
             )
 
         if tags:
-            matches = self._tags_to_filter(project, tags)
-            if not matches:
-                return queryset.none()
-            queryset = queryset.filter(
-                id__in=matches,
-            )
+            filter_tags = {k: v for k, v in six.iteritems(tags) if k[0] != '-'}
+            exclude_tags = {k[1:]: v for k, v in six.iteritems(tags) if k[0] == '-'}
+            if filter_tags:
+                matches = self._tags_to_filter(project, filter_tags)
+                if not matches:
+                    return queryset.none()
+                queryset = queryset.filter(
+                    id__in=matches,
+                )
+            if exclude_tags:
+                matches = self._tags_to_filter(project, exclude_tags)
```

Comment:
`_tags_to_filter` will [return at most `1000` group primary keys](https://github.com/getsentry/sentry/blob/master/src/sentry/search/django/backend.py#L62), so this will possibly  result in false positives making it through this exclusion filter depending on how many groups exist within a project. I'm unsure if this is a realistic issue or not. My personal opinion is that we shouldn't provide a search that's intentionally going to return incorrect results, but we already don't necessarily return the correct results with the inclusion filters, so I'm not really sure how to proceed here.